plugins/org.yakindu.sct.doc.user/src/user-guide/statechart_language.textile

```diff
@@ -627,3 +738,52 @@ bc. myBool = active(StateA)
+
+On its way, the train passes several "stations". Each station represents a language element that is valid at this point. In our example, the first station is a rectangle with a grey background. The grey background denotes a non-terminal, and the word inside the rectangle is the non-terminal's name, here: _ID_. You will have to look up this particular non-terminal's railroad diagram to learn more about it. However, for simplicity and because their meanings are more or less obvious a few non-terminals are not refined further. Examples are _STRING_, _INT_, _BOOL_, or _ID_.
+
+In a _SimpleElementReferenceExpression_, a valid ID – a name – is always the first element, e.&nbsp;g. @init42@. If the train turns right at the junction, it moves directly to the railroad diagram's exit. That means just an _ID_ like @init42@ is a valid _SimpleElementReferenceExpression_. Please note that when the train reaches the junction right before the exit, it cannot turn left, because trains don't move that way. That is, the train is unable to travel to the @(@ station from this direction.
+
+The train can reach the @)@ from the opposite direction only. After having left the initial _ID_ station, the train can move to the left at the junction and then run a stretch touching only the @(@ and @)@ stations, in that order. These rectangles have a white background color, denoting a "_terminal symbol_":https://en.wikipedia.org/wiki/Terminal_and_nonterminal_symbols or _terminal_ for short. A terminal stands for itself; write it down as it is.
+
+Following that rule, @init42()@ is a valid _SimpleElementReferenceExpression_. One or more _Expressions_ inside the paranthesis, separated by the @,@ terminal, are also valid. Example: @init42(foo, bar + 27)@.
+
+bq.. *Please note:*
+
+The railroad diagrams themselves are generated from a textual grammar representation established and maintained with "Xtext":https://eclipse.org/Xtext/. If you suspect the diagrams shown here are lagging behind the actual implementation, look for files with the @.xtext@ extension in the YAKINDU Statechart Tools source code distribution!
+
+bq.. *Please also note:*
+
+Not each and every construction that is syntactically allowed does make sense semantically. The statechart editor will flag such constructions as errors.
+
+h3(#statechart-grammar). Statechart grammar
+
+Figure "&quot;Statechart grammar&quot;":#fig-statechart-grammar shows the statechart grammar. It makes use of the "expressions grammar":#expressions-grammar.
+
+p(#fig-statechart-grammar). 
+
+!(standard-image)images/docu_grammar_010_stext.png(Statechart grammar)!
+
+p=. Statechart grammar
+
+h3(#expressions-grammar). Expressions grammar
+
+Figure "&quot;Expressions grammar&quot;":#fig-expressions-grammar shows the expressions grammar. It is used by the "statechart grammar":#statechart-grammar.
+
+p(#fig-expressions-grammar). 
+
+!(standard-image)images/docu_grammar_010_expressions.png(Expressions grammar)!
+
+p=. Expressions grammar
```

Comment:
I don't think the complete language grammar is really useful. The users who are really interested in its details because they want to adjust the language, would need to read the xtext grammar file.

I think it is best to extract some interesting parts of it and use it in the documentation (transition, state, statechart specification) instead of having the whole grammar on one shot.